src/app/static/src/js/utils/helpers.ts

```diff
@@ -43,4 +43,32 @@ export function workflowRunDetailsToMetadata(details: WorkflowRun): RunWorkflowM
     }
 }
 
+export function formatDate(date) {
+    return longTimestamp(new Date(date));
+}
+
+export const failStates = ["error", "orphan", "impossible", "missing", "interrupted"];
+export const notStartedStates = ["queued", "deferred", "impossible", "missing"];
+const nonFailStateMessages = {
+    "success": "Complete",
+    "impossible": "Dependency failed",
+    "deferred": "Waiting for dependency"
+};
+
+
+export function interpretStatus(status) {
+    if (Object.keys(nonFailStateMessages).includes(status)) {
+        return nonFailStateMessages[status]
+    } else if (
+        failStates.includes(status)
+    ) {
+        return "Failed";
+    } else {
+        return status.charAt(0).toUpperCase() + status.slice(1);
+    }
+}
```

Comment:
Could maybe make it a little more explicit that these refer to running report states and status (as opposed to workflow, or admin actions or whatever). Could either rename the individual consts e.g. `interpretReportStatus` or bundle these up into a separate object called `RunninReportStates` or something.